run-puppet

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -eu
+
+script=$(basename $0)
+usage="Usage:
+  $script [-n][-c <control-repo>]
+
+Notes:
+  root permissions are required. Best to run with 'sudo'
+  The environment variable CONTROL_REPO can be used in place of the '-c'
+    argument.
+"
+
+control_repo=${CONTROL_REPO:-}
+r10k_fetch=true
+puppet=/opt/puppetlabs/puppet/bin/puppet
+r10k=/opt/puppetlabs/puppet/bin/r10k
+
+function command_exists() {
+  type "$1" &> /dev/null ;
+}
+
+function install_puppet() {
+  if command_exists dpkg; then
+    # Currently only supports ubuntu
+    source /etc/os-release
+    codename=$(lsb_release --codename | cut --fields=2)
+    release_file="puppet-release-$codename.deb"
+    wget https://apt.puppetlabs.com/release_file
+    dpkg --install $release_file
+    rm $release_file
+    apt-get update
+    apt-get install puppet-agent --yes
+  elif command_exists yum; then
+    source /etc/os-release
+    release_file="puppet-release-el-$VERSION_ID.noarch.rpm"
+    rpm --upgrade --verbose --hash "https://yum.puppetlabs.com/puppet/$release_file"
+    yum install puppet-agent --assumeyes
+  else
+    echo "$script: ERROR Unsupported platform" 1>&2
+    exit 1
+  fi
+}
+
+function install_module() {
+  $puppet module install $1
+}
+
+function run_puppet() {
+  $r10k deploy environment -pv
+  $puppet apply \
+          --modulepath /etc/puppetlabs/code/environments/production/modules \
+          /etc/puppetlabs/code/environments/production/site.pp
+}
+
+while getopts ":c:hn" option; do
+  case $option in
+    c) control_repo=$OPTARG
+       ;;
+    h|\?) echo "$usage" 1>&2
+       exit 0
+       ;;
+    n) r10k_fetch=false
+       ;;
+    :) echo "$script: ERROR -$OPTARG requires an argument" 1>&2
+       exit 1
+       ;;
+  esac
+done
+shift $((OPTIND - 1))
+
+
+if [ $UID -ne 0 ]; then
+  echo "$script: ERROR This script must run as root to work properly. Use 'sudo'" 1>&2
+  exit 1
+fi
+
+if ! command_exists /opt/puppetlabs/puppet/bin/puppet; then
+  install_puppet
+fi
+
+if [ $r10k_fetch = true ]; then
```

Comment:
`$r10k_fetch` should really be named something like `$r10k_initialize` or something similar.

We are technically using r10k to fetch updated version of the modules every time we call `run_puppet` whereas here we are bootstrapping that environment.